File: API/Section/Export/C#/Export/ExportForm.cs
using GrapeCity.ActiveReports;
using GrapeCity.ActiveReports.Export;
using System;
using System.IO;
using System.Windows.Forms;

namespace Export
{
	public partial class ExportForm : Form
	{

		private const string _pdf = "pdf";
		private const string _xlsx = "xlsx";
		private const string _mht = "mht";
		private const string _rtf = "rtf";
		private const string _csv = "csv";

		public ExportForm()
		{
			InitializeComponent();
		}

		private void ExportForm_Load(object sender, EventArgs e)
		{
			reportsNames.SelectedIndex = 0;
			exportTypes.SelectedIndex = 0;
		}

		private void exportButton_Click(object sender, EventArgs e)
		{
			var reportName = (string)reportsNames.SelectedItem;
			var exportType = (string)exportTypes.SelectedItem;
			IDocumentExportEx documentExportEx = null;

			var report = new SectionReport();
			System.Xml.XmlTextReader xtr = new System.Xml.XmlTextReader(@"..\..\..\..\Reports\" + reportName);
			report.LoadLayout(xtr);
			report.Run();

			var exporTypeLower = exportType.ToLower();
			switch (exporTypeLower)
			{
				case _pdf:
					documentExportEx = new GrapeCity.ActiveReports.Export.Pdf.Section.PdfExport();
					break;
				case _xlsx:
					documentExportEx = new GrapeCity.ActiveReports.Export.Excel.Section.XlsExport(){ FileFormat = GrapeCity.ActiveReports.Export.Excel.Section.FileFormat.Xlsx};
					break;
				case _rtf:
					documentExportEx = new GrapeCity.ActiveReports.Export.Word.Section.RtfExport();
					break;
				case _mht:
					documentExportEx = new GrapeCity.ActiveReports.Export.Html.Section.HtmlExport();
					break;
				case _csv:
					documentExportEx = new GrapeCity.ActiveReports.Export.Xml.Section.TextExport();
					break;
			}

			var fileName = Path.GetFileNameWithoutExtension(reportName);
			var saveFileDialog = new SaveFileDialog()
			{
				FileName = fileName + GetSaveDialogExtension(exporTypeLower),
				Filter = GetSaveDialogFilter(exporTypeLower)
			};


			if (saveFileDialog.ShowDialog() != DialogResult.OK)
			{
				return;
			}

			exportButton.Enabled = false;
			using (var stream = new FileStream(Path.Combine(saveFileDialog.FileName), FileMode.Create))
				documentExportEx.Export(report.Document, stream);
			exportButton.Enabled = true;
		}

		private string GetSaveDialogFilter(string exportType)
		{
			return exportType + " files | *." + exportType;
		}

		private string GetSaveDialogExtension(string exportType)
		{
			return "." + exportType;
		}
	}
}

File: API/Section/Export/VB.NET/Export/ExportForm.vb
Imports System.IO
Imports GrapeCity.ActiveReports

Public Class ExportForm

	Const _pdf As String = "pdf"
	Const _xlsx As String = "xlsx"
	Const _mht As String = "mht"
	Const _rtf As String = "rtf"
	Const _csv As String = "csv"

	Private Sub exportButton_Click(sender As Object, e As EventArgs) Handles exportButton.Click
		Dim reportName = CType(reportsNames.SelectedItem, String)
		Dim exportType = CType(exportTypes.SelectedItem, String)
		Dim documentExportEx As GrapeCity.ActiveReports.Export.IDocumentExportEx = Nothing

		Dim report As SectionReport = New SectionReport()
		Dim xtr As New System.Xml.XmlTextReader("..\..\..\..\Reports\" + reportName)
		report.LoadLayout(xtr)
		report.Run()

		Dim exporTypeLower = exportType.ToLower()
		Select Case exporTypeLower
			Case _pdf
				documentExportEx = New GrapeCity.ActiveReports.Export.Pdf.Section.PdfExport()
				Exit Select
			Case _xlsx
				documentExportEx = New GrapeCity.ActiveReports.Export.Excel.Section.XlsExport() With {
					.FileFormat = GrapeCity.ActiveReports.Export.Excel.Section.FileFormat.Xlsx
				}
				Exit Select
			Case _rtf
				documentExportEx = New GrapeCity.ActiveReports.Export.Word.Section.RtfExport()
				Exit Select
			Case _mht
				documentExportEx = New GrapeCity.ActiveReports.Export.Html.Section.HtmlExport()
				Exit Select
			Case _csv
				documentExportEx = New GrapeCity.ActiveReports.Export.Xml.Section.TextExport()
				Exit Select
		End Select

		Dim fileName = Path.GetFileNameWithoutExtension(reportName)
		Dim saveFileDialog = New SaveFileDialog() With {
			.FileName = fileName + GetSaveDialogExtension(exporTypeLower),
			.Filter = GetSaveDialogFilter(exporTypeLower)
		}

		If saveFileDialog.ShowDialog() <> DialogResult.OK Then
			Return
		End If

		exportButton.Enabled = False
		Using stream As FileStream = New FileStream(saveFileDialog.FileName, FileMode.Create)
			documentExportEx.Export(report.Document, stream)
		End Using
		exportButton.Enabled = True
	End Sub

	Private Sub ExportForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
		reportsNames.SelectedIndex = 0
		exportTypes.SelectedIndex = 0
	End Sub

	Private Function GetSaveDialogFilter(exportType As String) As String
		Return exportType + " files | *." + exportType
	End Function

	Private Function GetSaveDialogExtension(exportType As String) As String
		Return "." + exportType
	End Function

End Class
